server/knex/migrations/20200505210059_create_users_table.js

```diff
@@ -0,0 +1,46 @@
+var bcrypt = require('bcryptjs');
+
+exports.up = function(db, Promise) {
+    return db.schema.hasTable('users').then(function(exists) {
+        if (!exists) {
+            return db.schema.createTable('users', table => {
+                table.charset('utf8');
+                table.collate('utf8_general_ci');
+                table.increments('id').primary().unique().notNullable();
+                table.string('givenname', [255]).notNullable();
+                table.string('surname',[255])
+                table.text('username').notNullable().unique();
+                table.string('password').notNullable()
+                table.string('email').notNullable().unique();
+                table.enu('role', ['admin', 'user']).notNullable().defaultTo('user')
+                table.enu('status', ['active', 'disabled']).notNullable().defaultTo('disabled')
+                table.datetime('lastlogondate').notNullable();
+            })
+            .then(function (){
+              //create the default admin user
+              const salt = bcrypt.genSaltSync();
+              const hash = bcrypt.hashSync('changeme', salt);
```

Comment:
`const hash = bcrypt.hashSync('changeme', salt);`

should the migration not import the existing hashed pw, or incompatible formats? I could see a lot of accidental wide open instances where people upgrade without thinking

Reply:
It should, i stupidly didn't think of just copying the existing hash from the config file.... lol 

I could probably import the username too huh.... Another one for the list.